app/models/discount.rb

```diff
@@ -0,0 +1,19 @@
+class Discount
+  def self.apply(line_item)
+    discounts(line_item)
+  end
+
+  def self.discounts(item)
+    { "PC" => pepsi_cola(item), "CC" => coca_cola(item) }[item.code]
+  end
+
+  def self.pepsi_cola(item)
+    (item.price * item.quantity) * 0.80 if item.quantity >= 3
+  end
+
+  def self.coca_cola(item)
+    (item.price * item.quantity) - item.price if item.quantity >= 2
+  end
+
+  private_class_method :discounts, :pepsi_cola, :coca_cola
+end
```

Comment:
Here I've considered serval options.

The first being what we have here which is a singleton class responsible for take a line item and determining if any discounts can be applied to the given item's total. The discount data is hard-coded for ease of access and readability.

The second option would be to have a similar architecture but instead accessing the discount reference and logic via the database. That would be a more scalable solution in the long term since we could theoretically have many discount for many scenarios. To actually store the logic as is in a database would take some clever changes in how we consume the given discount data.

For the reasons mentioned above I went ahead with the more explicit and and cleaner solution, worrying about scalability when we faced with that theoretical scenario.